File: app/types/reports.ts
export type ReportStatus =
  | "Ready"
  | "Processing"
  | "Needs Review"
  | "Failed"
  | string;

export type ReportRow = {
  id: string;
  company_id?: string;

  report_name: string;
  module: string;
  period: string;

  status: ReportStatus;
  last_generated: string;

  file_url?: string;
  created_at?: string;
};

export type ReportStats = {
  reports_today: number;
  ready: number;
  processing: number;
  needs_review: number;
};

File: backend/app/schemas/reports.py
from pydantic import BaseModel
from typing import Optional


class ReportStats(BaseModel):
    reports_today: int
    ready: int
    processing: int
    needs_review: int


class ReportRow(BaseModel):
    id: str
    company_id: str
    report_name: str
    module: str
    period: str
    status: str
    last_generated: str
    file_url: Optional[str] = None
    created_at: str